server/vulnerabilities/nvd/cve.go

```diff
@@ -654,29 +654,36 @@ func buildConstraintString(startIncluding, startExcluding, endExcluding string)
 }
 
 // Products using 4 part versioning scheme (ie. docker desktop)
-// need to be converted to 3 part versioning scheme (2.3.0.2 -> 2.3.0+3) for use with
+// need to be converted to 3 part versioning scheme (2.3.0.2 -> 2.3.0-3) for use with
 // the semver library.
 func preprocessVersion(version string) string {
-	// If "+" is already present, validate the part before "+" as a semver
-	if strings.Contains(version, "+") {
-		parts := strings.Split(version, "+")
+	// If "-" is already present, validate the part before "-" as a semver
```

Comment:
If it already has `-` isn't the whole thing semver compatible? IOW do you need to split?